### PR TITLE
Added Roberta model with base and large configs for question answering

### PR DESCRIPTION
**Pull request by jimypbr**
Roberta now running on IPU :)

Tested base and large on squad v1.

Base:
```
wandb:                       train/loss 0.6118
wandb:              train/learning_rate 0.0
wandb:                      train/epoch 3.0
wandb:                train/global_step 516
wandb:                         _runtime 458
wandb:                       _timestamp 1639156425
wandb:                            _step 52
wandb:              train/train_runtime 268.7952
wandb:   train/train_samples_per_second 988.5
wandb:     train/train_steps_per_second 1.92
wandb:                 train/total_flos 6.627121224537342e+18
wandb:                 train/train_loss 1.0056
wandb:                 eval/exact_match 85.75213
wandb:                          eval/f1 92.08115
```

Large:
```
wandb:                       train/loss 0.5226
wandb:              train/learning_rate 0.0
wandb:                      train/epoch 3.0
wandb:                train/global_step 261
wandb:                         _runtime 983
wandb:                       _timestamp 1639396371
wandb:                            _step 27
wandb:              train/train_runtime 636.8869
wandb:   train/train_samples_per_second 417.192
wandb:     train/train_steps_per_second 0.41
wandb:                 train/total_flos 9.531261908893355e+19
wandb:                 train/train_loss 0.97289
wandb:                 eval/exact_match 88.69442
wandb:                          eval/f1 94.4119
```

### CODE REVIEW

File: optimum/graphcore/models/roberta/modeling_roberta.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        return self
+
+    def forward(self, input_ids, attention_mask, start_positions=None, end_positions=None):
+        output = super().forward(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            start_positions=start_positions,
+            end_positions=end_positions,
+        )
+        if self.training:
```
Comment (michaelbenayoun) [inline optimum/graphcore/models/roberta/modeling_roberta.py:183]: To better integrate with `transformers` and especially with the examples, I think we should match the way we return the outputs and the way [it's being done](https://github.com/huggingface/transformers/blob/master/src/transformers/models/roberta/modeling_roberta.py#L1539) for `RobertaForQuestionAnswering`. 

Basically choosing to return a loss if both `start_positions` and `end_positions` are provided.

**Review by michaelbenayoun — APPROVED** [review]

**Comment by michaelbenayoun** [comment]
Nice work @jimypbr !
